metadata-load/src/main/java/uk/ac/ebi/ampt2d/metadata/importer/ObjectsImporter.java

```diff
@@ -142,9 +162,64 @@ public Analysis importAnalysis(String accession) {
+        ReferenceSequence referenceSequence = null;
+        try {
+            String assemblyXml = sraXmlRetrieverByAccession.getXml(accession);
+            AssemblyType assembly = sraAssemblyXmlParser.parseXml(assemblyXml, accession);
+            referenceSequence = referenceSequenceConverter.convert(assembly);
+            Taxonomy taxonomy = taxonomyExtractorFromReferenceSequence.getTaxonomy(assembly.getTAXON());
+            referenceSequence.setTaxonomy(taxonomy);
+            referenceSequence = referenceSequenceRepository.findOrSave(referenceSequence);
+        } catch (Exception exception) {
+            IMPORT_LOGGER.log(Level.SEVERE, "Encountered Exception for ReferenceSequence accession " + accession);
+            IMPORT_LOGGER.log(Level.SEVERE, exception.getMessage());
+        }
+        return referenceSequence;
+    }
+
+    private Set<String> getReferenceSequenceAccessions(AnalysisType analysisType) {
+        Set<String> referenceSequenceAccessions = new HashSet<>();
+        AnalysisType.ANALYSISTYPE analysisType1 = analysisType.getANALYSISTYPE();
+        // ENA records can contain a reference sequence in either of the three analysis categories:
+        // REFERENCE_ALIGNMENT, SEQUENCE_VARIATION, and PROCESSED_READS. All accessions are collected and returned.
+        if (analysisType1.isSetREFERENCEALIGNMENT()) {
+            String accession = getAccessionFromReferenceSequenceType(analysisType1.getREFERENCEALIGNMENT());
+            if (accession != null) {
+                referenceSequenceAccessions.add(accession);
+            }
+        }
+        if (analysisType1.isSetSEQUENCEVARIATION()) {
+            String accession = getAccessionFromReferenceSequenceType(analysisType1.getSEQUENCEVARIATION());
+            if (accession != null) {
+                referenceSequenceAccessions.add(accession);
+            }
+        }
+        if (analysisType1.isSetPROCESSEDREADS()) {
+            String accession = getAccessionFromReferenceSequenceType(analysisType1.getPROCESSEDREADS());
+            if (accession != null) {
+                referenceSequenceAccessions.add(accession);
+            }
+        }
+        return referenceSequenceAccessions;
+    }
+
+    private String getAccessionFromReferenceSequenceType(ReferenceSequenceType referenceSequenceType) {
+        if (referenceSequenceType != null) {
+            ReferenceAssemblyType referenceAssemblyType = referenceSequenceType.getASSEMBLY();
+            if (referenceAssemblyType != null) {
+                ReferenceAssemblyType.STANDARD standard = referenceAssemblyType.getSTANDARD();
+                if (standard != null) {
+                    return getAccessionFromStandard(standard);
+                }
+            }
+        }
         return null;
     }
 
+    protected String getAccessionFromStandard(ReferenceAssemblyType.STANDARD standard) {
+        return standard.getAccession();
```

Comment:
@cyenyxe just wanna confirm this is fine? 
[ObjectImporter::getAccessionFromStandard() and SraObjectsImporterThroughDatabase::getAccessionFromStandard()   https://github.com/EBIvariation/metadata-ws/pull/50#discussion_r288505075 ]

Reply:
Yes, it must always return the accession.

Reply:
In the current implementation accession is always returned. Marking as resolved.